builtInFunctions/esdtLocalMint.go

```diff
@@ -80,7 +82,7 @@ func (e *esdtLocalMint) ProcessBuiltinFunction(
 	}
 
 	tokenID := vmInput.Arguments[0]
-	err = e.rolesHandler.CheckAllowedToExecute(acntSnd, tokenID, []byte(core.ESDTRoleLocalMint))
+	err = e.isAllowedToExecute(acntSnd, tokenID, []byte(core.ESDTRoleLocalMint))
```

Comment:
I think it would be better to have the crossChainChecker used only by the rolesHandler. It is simpler. And we can have the same check for BURN as for others. no need to complicate.

Reply:
Right, that's a good suggestion. Refactored.